virtcontainers/sandbox.go

```diff
@@ -781,12 +799,18 @@ func createSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Fac
 	}
 
 	// Below code path is called only during create, because of earlier check.
-	if err := s.agent.createSandbox(s); err != nil {
+	defer func() {
+		if err != nil {
+			rollbackFailingSandboxCreation(ctx, s)
+		}
+	}()
+
+	if err = s.agent.createSandbox(s); err != nil {
```

Comment:
`s.agent.createSandbox` should be called before since your rollback function includes the agent cleanup, right?

Reply:
updated~